cmd/awsDataGather/main.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/red-hat-storage/ocs-osd-deployer/pkg/aws"
+	"github.com/red-hat-storage/ocs-osd-deployer/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+func main() {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	log := ctrl.Log.WithName("main")
+
+	namespace, found := os.LookupEnv("NAMESPACE")
+	if !found {
+		fmt.Fprintf(os.Stderr,
+			"NAMESPACE environment variable not found\n")
+		os.Exit(1)
+	}
+	podName, found := os.LookupEnv("POD_NAME")
+	if !found {
+		fmt.Fprintf(os.Stderr,
+			"POD_NAME environment variable not found\n")
+		os.Exit(1)
+	}
+
+	// We need the deployment name to use as an owner reference.
+	// However, the deployment name cannot be passed as an environment variable,
+	// so we derive it from the pod name.
+	deploymentName, err := utils.DeploymentNameFromPodName(podName)
+	if err != nil {
+		fmt.Fprintf(os.Stderr,
+			"Could not determine deployment name from pod name (%s)",
+			podName)
+		os.Exit(1)
+	}
+
+	log.Info("Setting up k8s client")
+	var options client.Options
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(clientgoscheme.AddToScheme(options.Scheme))
+
+	k8sClient, err := client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		log.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	// This will later be used as the OwnerReference for the data ConfigMap
+	var deployment appsv1.Deployment
+	err = utils.Retry(5, 1*time.Second, func() error {
```

Comment:
After checking my notes, I can't find a reason for adding this. My guess was I put this in just in case some network hiccup caused the get call to fail. I'll remove it